commands/patch_notes.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+const http = require('http');
+
+// splits a string into an array of `size` length characters
+function chunkString(str, size) {
+    const numChunks = Math.ceil(str.length / size),
+        chunks = new Array(numChunks);
+
+    for (let i = 0, o = 0; i < numChunks; ++i, o += size) {
+        chunks[i] = str.substr(o, size);
+    }
+
+  return chunks;
+}
+
+// TODO: should prevent breaking words and Markdown
+// takes a long message (>2000 chars) and breaks it down and sends it as multiple smaller messages
+function sendChunkedMessage(bot, channel, message) {
+    let chunks = chunkString(message, 2000);
+
+    if (chunks.length) {
+        bot.sendMessage(channel, chunks.shift(), err => {
+            if (!err) {
+                sendChunkedMessage(bot, channel, chunks.join(''));
+            } else {
+                console.log(err);
+            }
+        });
+    }
+}
+
+// performs a basic http GET
+function doHttpGet(url, done) {
+    http.get(url, res => {
+        let data = '';
+
+        res.on('data', d => {
+            data += d;
+        });
+
+        res.on('end', () => {
+            done(data);
+        });
+    });
+}
+
+// supported games
+const SUPPORTED_GAMES = {
+    // hots: {
+    //     displayName: 'Heroes of the Storm',
+    //     listingUrl: 'http://us.battle.net/heroes/en/search?f=article&k=Patch%20Notes&sort=time&dir=d',
+    //     scrape(doneCallback) {
+    //
+    //     }
+    // },
+    overwatch: {
+        displayName: 'Overwatch',
+        listingUrl: 'http://overwatch.wikia.com/api/v1/Articles/List?category=Patch+notes',
+        contentsUrl: 'http://overwatch.wikia.com/api/v1/Articles/AsSimpleJson?id=',
+        fetch(options, done) {
+            const HEADING_LEVELS = ['__**%s**__', '***%s***', '**%s**'],
+                listingCallback = (data) => {
+                    const listings = JSON.parse(data),
+                        patches = listings.items.filter(item => {
+                                return item.title !== 'List of Patches';
+                            }).sort((a, b) => {
+                                const regex = /^([\w,\s]*)(?:\(beta\))?$/,
+                                    aMatches = a.title.match(regex),
+                                    bMatches = b.title.match(regex),
+                                    aDate = new Date(aMatches[1]),
+                                    bDate = new Date(bMatches[1]);
+
+                                return aDate - bDate;
+                            }).reverse(),
+                        patch = patches[options.patchesAgo];
```

Comment:
If you supply an integer offset, e.g., `!patch_notes overwatch 1` it'll grab notes for that many patches *ago*, default value being `0`.